homeassistant/components/switch/broadlink.py

```diff
@@ -0,0 +1,261 @@
+"""
+Support for Broadlink RM devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.broadlink/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME, CONF_SWITCHES, CONF_COMMAND_OFF,
+    CONF_COMMAND_ON, CONF_OPTIMISTIC, CONF_HOST, CONF_MAC)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pycrypto==2.6.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Broadlink switch'
+
+SWITCH_SCHEMA = vol.Schema({
+    vol.Optional(CONF_COMMAND_OFF, default=None): cv.string,
+    vol.Optional(CONF_COMMAND_ON, default=None): cv.string,
+    vol.Optional(CONF_FRIENDLY_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_OPTIMISTIC, default=True): cv.boolean,
```

Comment:
I have one of these devices, and to my knowledge it doesn't keep state. Maybe this component was adapted from one that does?